GitFyle.Core.Api/Brokers/Storages/StorageBroker.Contributions.References.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using GitFyle.Core.Api.Models.Foundations.Contributions;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace GitFyle.Core.Api.Brokers.Storages
+{
+    internal partial class StorageBroker
+    {
+        void AddContributionConfigurations(EntityTypeBuilder<Contribution> builder)
+        {
+            builder.HasIndex(contribution => new
+                {
+                    contribution.RepositoryId,
+                    contribution.ExternalId
+                })
+                .IsUnique();
+
+            builder.Property(contribution => contribution.ExternalId)
+                .HasMaxLength(255)
+                .IsRequired();
+
+            builder.Property(contribution => contribution.Title)
+                .HasMaxLength(255)
+                .IsRequired();
+
```

Comment:
Define the relationships between Contibution and Repository, Contributor, ContributionType

1) You will need to add `public IEnumerable<Contribution> Contributions { get; set; }` to the Repository.cs, Contributor.cs and ContributionType.cs
2) Add these to Contribution.cs

```cs
        public Repository Repository { get; set; }
        public Contributor Contributor { get; set; }
        public ContributionType ContributionType { get; set; }
```

3) Then in this configuration file you can define your one-to-many relationship like this:
```cs
            modelBuilder.Entity<Contribution>()
                .HasOne(contribution => contribution.Repository)
                .WithMany(respository => respository.Contributions)
                .HasForeignKey(contribution => contribution.RepositoryId)
                .OnDelete(DeleteBehavior.NoAction);

            modelBuilder.Entity<Contribution>()
                .HasOne(contribution => contribution.Contributor)
                .WithMany(contributor => contributor.Contributions)
                .HasForeignKey(contribution => contribution.ContributorId)
                .OnDelete(DeleteBehavior.NoAction);

            modelBuilder.Entity<Contribution>()
                .HasOne(contribution => contribution.ContributionType)
                .WithMany(contributionType => contributionType.Contributions)
                .HasForeignKey(contribution => contribution.ContributionTypeId)
                .OnDelete(DeleteBehavior.NoAction);
```
4) Add your migration changes

Reply:
> do I need to wait for Repository, Contributor, ContributionType to be merged?

If you pull in the latest code from main, then you will have access to `ContributionType`

Just follow the instructions above to complete the relationship configuration to `ContributionType` only.   Once `Contributor` and `Repository` is merged, we can do another `DATA` pull request to add relationship configuration for the the other two.